### PR TITLE
Add /workspaces/* path into universal_link

### PR DESCRIPTION
**Pull request by studentofcoding**
### Details

**The problem is only appearing on Safari Mobile, with the detail of**

Workspaces - Visual flickering when navigating back to workspace list from a WS 

**Some extra note:** 

- The scope is on Workspace related page, but we also include `/save-the-world/*` which is outside of the scope of the issues
- The Screenshots are Expected results to be (on mobile safari section) since the Smart App Banner only shows on the production code.
- It seems like on the Save the World page, there is a small bug (different background color) on the SafeArea on top, if that is the case, a bug report has already been sent [here](https://expensify.slack.com/archives/C049HHMV9SM/p1693506283258809) & just want to note this, to avoid regression after it's merged

photo_2023-09-01 01 08 29

### Fixed Issues

$ [24341](https://github.com/Expensify/App/issues/24341)
PROPOSAL: https://github.com/Expensify/App/issues/24341#issuecomment-1678003081

### Tests

1. Open the staging weblink
2. Go to Settings > Workspaces
3. Select a workspace to open WS settings
4. Go back to workspace list by tapping on the back button
5. See that the Smart App Banner is also shown on Workspace Page Detail and the layout is fixed
6. 

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open the staging weblink
2. Go to Settings > Workspaces
3. Select a workspace to open WS settings
4. Go back to workspace list by tapping on the back button
5. See that the Smart App Banner is also shown on Workspace Page Detail and the layout is fixed

### QA Steps

1. Open the staging weblink
2. Go to Settings > Workspaces
3. Select a workspace to open WS settings
4. Go back to workspace list by tapping on the back button
5. See that the Smart App Banner is also shown on Workspace Page Detail and the layout is fixed

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

- Workspace

:workspace:*

- Save the world

:save-the-world:*

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by studentofcoding** [comment]
> @studentofcoding How am I supposed to test this on my localhost? Can you please update the test steps?

From my understanding, the Smart App Banner can only be shown on the production/staging site we have, so It'll behave the same as another path as it's using the same structure and no other changes are introduced so in this case 

If we want to wait for an extra explanation, I've recently opened a discussion about this on Slack and tagged you @allroundexperts 

Do you maybe have some wisdom on this @dangrous 

Note: I've tried deploying & setting up Local and Netlify but always got blocked on both 
```
GET https://www.expensify.com/api?command=Ping net::ERR_FAILED 200
192.168.1.3/:1 Access to fetch at 'https://www.expensify.com/api?command=Ping' from origin 'http://192.168.1.3:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

- Tried Localhost via CORS disabled, but it still can't be accessed

- Tried Netlify configuration with Netlify server https://expensify-yonathan.netlify.app/ + _headers config (/*
  Access-Control-Allow-Origin: *), but it still can't be accessed

**Comment by dangrous** [comment]
I'm updating the reviewers on this, not sure why it didn't pick us properly. Ignore this issue, @youssef-lr !

File: .well-known/apple-app-site-association
Comment (dangrous) [inline .well-known/apple-app-site-association:73]:
```suggestion
                        "comment": "Save the World"
```
I think this is fine, since we're going to be using these for different initiatives

**Review by dangrous — CHANGES_REQUESTED** [review]
Two edits on the comments!

Do we know of a way to actually test this? Now that we can make different test builds would any of those work, maybe? cc @Julesssss since you worked on the test builds thing

**Comment by studentofcoding** [comment]
> Two edits on the comments!
> 
> Do we know of a way to actually test this? Now that we can make different test builds would any of those work, maybe? cc @Julesssss since you worked on the test builds thing

Done it @dangrous, also This is our last discussion with Aswin on `apple-app-site-association` 

https://expensify.slack.com/archives/C01GTK53T8Q/p1693544190232139?thread_ts=1693543888.454649&cid=C01GTK53T8Q

**Comment by Julesssss** [comment]
> Two edits on the comments!
> 
> Do we know of a way to actually test this? Now that we can make different test builds would any of those work, maybe? cc @Julesssss since you worked on the test builds thing

I kicked off a test build job for you [here](https://github.com/Expensify/App/actions/runs/6099188638). A comment should appear shortly with links to all platforms

**Comment by studentofcoding** [comment]
iOS build failed @Julesssss @dangrous (probably because the Branch isn't synced as Master Branch)

Also upon checking on my iPhone for the web, the Smart App Banner didn't show on all the pages, it probably this is related to how it only load on the listed URL here :

Screenshot 2023-09-07 at 00 25 11

**Comment by dangrous** [comment]
Ah okay. I thinkkkk it'll probably be fine? It's definitely not going to break anything, the question is whether or not it will actually work once deployed, haha. So maybe we just push (when the freeze is off) with prod QA, and handle any issues at that point.

**Review by dangrous — APPROVED** [review]

**Comment by studentofcoding** [comment]
> Ah okay. I thinkkkk it'll probably be fine? It's definitely not going to break anything, the question is whether or not it will actually work once deployed, haha. So maybe we just push (when the freeze is off) with prod QA, and handle any issues at that point.

Yess, I also think that it'll finee. This is actually intriguing as we only can make sure after its being deployed later lol

Meanwhile, I'll re-check and add another configuration (if needed)

**Comment by studentofcoding** [comment]
As the conferences are done, I think its safe to continue @mountiny ? 

Also waiting for your approval @allroundexperts 

Thanks!

**Comment by studentofcoding** [comment]
Also Do I need to merge main to this PR branch @dangrous

**Comment by mountiny** [comment]
@studentofcoding yes please, merge main and retest please. Thanks!

**Comment by studentofcoding** [comment]
Main is merged @dangrous @allroundexperts, It's ready to be merged to main & tested there!

**Comment by studentofcoding** [comment]
Can we remove the HOLD merge freeze @mountiny ? Thanks

**Comment by dangrous** [comment]
I think this is good to go, I'm just checking to see what the process is for having this tested on prod instead of on staging

**Comment by studentofcoding** [comment]
> I think this is good to go, I'm just checking to see what the process is for having this tested on prod instead of on staging

Noted, I'll standby in case any extra conf needed from my part

**Review by allroundexperts — APPROVED** [review]
Let's 🚀 it!

**Comment by studentofcoding** [comment]
Lets merged itt! 

cc: @dangrous @allroundexperts

**Comment by dangrous** [comment]
yes please!

**Comment by studentofcoding** [comment]
Also added additional context [here](https://github.com/Expensify/App/issues/24341#issuecomment-1720009848) @dangrous @kavimuru as Smart App Banner is showing fine on Staging while on Production it isn't, and this is my hunch on it

> My hunch is this behavior is happening because the Production still didn't have the last updated change from Staging, which is the change from internal PR [below](https://github.com/Expensify/App/commit/10c41da000bb36d46da014111921b04ddadae50d) (that basically change /save-the-world/* into /teachersunite/* that make the Smart App Banner didn't behave as expected

I've poked Aswin as well, hopefully we can get more insight on it 😄

**Pull request merged**
